src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -1029,6 +1029,39 @@ void Test12()
             VerifyNotInScope(model, x12Ref[1]);
         }
 
+        [Fact]
+        public void PropertyNamedInComplexPattern()
+        {
+            var source =
+@"
+using System;
+public class Program
+{
+    public static void Main()
+    {
+        object o = nameof(Main);
+        Console.WriteLine(o is string { Length is 4 });
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
+
+            compilation.VerifyDiagnostics();
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+            var propPats = tree.GetRoot().DescendantNodes().OfType<SubPropertyPatternSyntax>().ToArray();
+            foreach (var p in propPats)
+            {
+                var si = model.GetSymbolInfo(p);
+                Assert.NotNull(si.Symbol);
+                Assert.Equal("Length", si.Symbol.Name);
+                si = model.GetSymbolInfo(p.Left);
+                Assert.NotNull(si.Symbol);
+                Assert.Equal("Length", si.Symbol.Name);
+            }
+
+        }
```

Comment:
There is already a test for SubPropertyPatternSyntax.Left passed to GetSymbol info (line 1058).
There is no overload of GetSymbolInfo that takes an ExpressionSyntax, and only p.Left is already an ExpressionSyntax.
I will add an error scenario test.